src/rgw/rgw_rest_s3.cc

```diff
@@ -4878,10 +4878,12 @@ RGWHandler_REST* RGWRESTMgr_S3::get_handler(rgw::sal::RGWRadosStore *store,
   } else {
     if (s->init_state.url_bucket.empty()) {
       handler = new RGWHandler_REST_Service_S3(auth_registry, enable_sts, enable_iam, enable_pubsub);
-    } else if (rgw::sal::RGWObject::empty(s->object.get())) {
-      handler = new RGWHandler_REST_Bucket_S3(auth_registry, enable_pubsub);
-    } else {
+    } else if (!rgw::sal::RGWObject::empty(s->object.get())) {
       handler = new RGWHandler_REST_Obj_S3(auth_registry);
+    } else if (s->info.args.exist_obj_excl_sub_resource()) {
+      return NULL;
+    } else {
+      handler = new RGWHandler_REST_Bucket_S3(auth_registry, enable_pubsub);
```

Comment:
Request like "DELETE bucket?requestPayment=aaaaa", have you tested it? I found it can also delete the empty bucket, and requestPayment is not the obj_excl_sub_resource. @BryceCao @ofriedma

Reply:
requestPayment is  bucket_exclusive, and request like "DELETE bucket?requestPayment=aaaaa" is a scene beyond this PR, its even not a S3 api. Maybe we can make another PR to discuss and save this problem.

Reply:
@BryceCao 
#38731 is a complete solution for this bug, and it can handle the situation I mentioned above. So it's up to @ofriedma